From 3069700d0efcfe506a18c4464ca6c761fa30bbad Mon Sep 17 00:00:00 2001
From: David Brazdil <dbrazdil@google.com>
Date: Mon, 5 Feb 2018 20:04:16 +0000
Subject: [PATCH 097/306] Exempt system apps from hidden API warnings

Bug: 64382372
Test: make
Change-Id: Ib9bb92f29858dea4f53b0b1ba3b40bdf22b897c9
(cherry picked from commit 9e93d57221bb7b8feb82d3ccf0896f5fa4f13a81)
---
 core/java/android/content/pm/ApplicationInfo.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/android/content/pm/ApplicationInfo.java b/core/java/android/content/pm/ApplicationInfo.java
index b61a6d997b1..db1630b67bd 100644
--- a/core/java/android/content/pm/ApplicationInfo.java
+++ b/core/java/android/content/pm/ApplicationInfo.java
@@ -1601,7 +1601,7 @@ public class ApplicationInfo extends PackageItemInfo implements Parcelable {
      * @hide
      */
     public boolean isAllowedToUseHiddenApi() {
-        return false;
+        return isSystemApp();
     }
 
     /**
-- 
2.17.1

